src/core/lib/surface/completion_queue.h

```diff
@@ -49,6 +49,8 @@ extern int grpc_trace_pending_tags;
 #endif
 
 typedef struct grpc_cq_completion {
+  gpr_mpscq_node node;
```

Comment:
Yes. It can be but we still need extra storage to store success..
so something like 
``` C
  typedef struct grpc_cq_completion {
    union {
      gpr_mpscq_node node;
      unitptr_t next;
      bool is_success;
    } list_data;
   }
```

Did you have something else in mind ?